WordPress/gutenberg
# Site Editor: update headings hierarchy in the 'Manage all' screens

## What?
Resolves https://github.com/WordPress/gutenberg/issues/52254

This PR:

- updates heading hierarchy on the template and template part pages
- passes a `level` prop to Header from Page component

## Why?
The heading hierarchy was incorrect in that there were two h1s and, underneath, multiple h5s. There should be a incremental order.

## Testing Instructions

Taken from https://github.com/WordPress/gutenberg/issues/52254

1. Use the HeadingsMap browser extension.
2. Go to the **Site editor > Design > Templates > Manage all templates**.
3. Check that the headings hierarchy is correct (h1 > h2 > h3).
4. Go to the **Site editor > Design > Patterns > Manage all template parts**.
5. Observe the headings  hierarchy is correct (h1 > h2 > h3).

Compare with trunk - the font size should be pretty close.



## Screenshots or screencast 


<img width="644" alt="Screenshot 2023-07-05 at 9 04 29 am" src="https://github.com/WordPress/gutenberg/assets/6458278/5d4f9f32-1c6c-472e-bb69-fe09ae722a2f">
<img width="722" alt="Screenshot 2023-07-05 at 9 04 17 am" src="https://github.com/WordPress/gutenberg/assets/6458278/fd7142ec-9ac4-4da3-853a-d1b1b1950546">
<img width="1060" alt="Screenshot 2023-07-05 at 9 04 04 am" src="https://github.com/WordPress/gutenberg/assets/6458278/5e4c672a-dbaf-4251-ad34-9c92487489cc">
